R/namespace_linter.R

```diff
@@ -85,9 +85,14 @@ namespace_linter <- function(check_exports = TRUE, check_nonexports = TRUE) {
     # run here, not in the factory, to allow for run- vs. "compile"-time differences in package structure
     namespaces <- lapply(packages, function(package) tryCatch(getNamespace(package), error = identity))
     failed_namespace <- vapply(namespaces, inherits, "condition", FUN.VALUE = logical(1L))
+    # nocov start
     if (any(failed_namespace)) {
-      stop("Failed to retrieve namespaces for one or more of the packages. Please report this issue.") # nocov
+      stop(
+        "Failed to retrieve namespaces for one or more of the packages. ",
```

Comment:
Let's improve this message too while we're at it -- "the packages" is a bit vague, what packages?

We could either describe it generally "one or more of the packages used with `::` or `:::`", or more specifically by actually naming the packages.

Reply:
For now, I have just included the generic message.

I have created a new issue (https://github.com/r-lib/lintr/issues/1855) to remember to improve the message by including problematic package names, since it isn't as trivial as I thought, and don't want to expand the scope of this PR beyond what I originally had in mind.